### ISSUE TITLE
Bug in SVN is_pristine 

### ISSUE THREAD

**Issue by keithrob91** [conan-io/conan#4628]
Conan 1.12.3

The user warning regarding `is_pristine` not being implemented on svn versions prior to 1.9 is a little confused! See below when I use it with svn 1.7.14:

```
UserWarning: SVN::is_pristine for SVN v1.7.14 (less than 1.9) is not implemented, it is returning not-pristine always because it cannot compare with checked out version.
  " checked out version.".format(self.version, SVN.API_CHANGE_VERSION))
```

**Comment by jgsogo** [comment]
Hi, @keithrob91, why is it confusing? That functionality is implemented from SVN version `v1.9` onwards (where there it is an API change in the SVN commands)

**Comment by keithrob91** [comment]
It was the following text I was referring to:
`.format(self.version, SVN.API_CHANGE_VERSION))`

i.e. did you really mean for that to be bolted onto the end?

**Comment by keithrob91** [comment]
Sorry, I meant:
` " checked out version.".format(self.version, SVN.API_CHANGE_VERSION))`

Is that what you wanted bolted onto the end?

**Comment by jgsogo** [comment]
Oh! I see, sorry for not realizing about it before. Let's see if we can do something for the next release to fix it, it should be quite easy.

By default the `warnings` module prints the message and the line of code in the source file too.
The documentation about the `warnings` module says for the function `showwarning` ([link](https://docs.python.org/3/library/warnings.html#warnings.showwarning)): _"You may replace this function with any callable by assigning to warnings.showwarning"_, so something like:

```python
import warnings

def just_the_message(msg, *args, **kwargs):
    return str(msg) + '\n'

warnings.showwarning = just_the_message
warnings.warn("SVN::is_pristine....")
```

Probably it should be a context manager to restore the previous function (there are more warnings in the codebase) or maybe we should configure it for all the Conan client lifetime.

Let's enqueue it

**Comment by keithrob91** [comment]
I know this is a (very) minor issue, but when is it planned to be fixed?

**Comment by jgsogo** [comment]
Hi! I've just tested it and in Conan v1.14 the warning is printed properly, can you try it? Thanks!

**Comment by keithrob91** [comment]
Looks good, thank you.

**Comment by jgsogo** [comment]
It was fixed thanks to changes in #4548, I didn't link to this issue from there.

Thanks!
